### ISSUE TITLE
Feature request: add wildcard support into Split tunneling

### ISSUE THREAD

**Issue by niki4xxxxx** [amnezia-vpn/amnezia-client#856]
**Describe the issue**
When wildcards like *.web.site are used, the program does not forward traffic that goes to subdomains when Split tunneling is used.

**To Reproduce**

1. Find a website that's blocked in your country (e.g. Zendesk in RF)
2. Enable Split tunneling to use VPN for only selected websites
3. Add *.zendesk.com
4. Enable VPN
5. Open https://support.zendesk.com/hc/en-us 

**Expected behavior**
All subdomains of zendesk.com can be opened.

**Desktop (please complete the following information):**
 - OS: Kubuntu 22.04 (Kernel 6.5.0-35)
 - Client: 4.5.3.0 (Apr 30 2024, eb48eb4)

**Server (please complete the following information):**
 - OS: Ubuntu 20.04

**Additional context**
The subdomain can be opened when it is specified in the Split tunneling settings, e.g. I can open https://support.zendesk.com/hc/en-us if I add support.zendesk.com

**Comment by Nexen23** [comment]
Really wanted that feature on windows so tried to see if I can implement that, but got stuck in the beggining. 
Amnezia internally uses something called [MIB_IPFORWARDTABLE](https://learn.microsoft.com/en-us/windows/win32/api/ipmib/ns-ipmib-mib_ipforwardrow#members) which accepts only IPs. When user adds host to split tunnelling, it is converted to IP and then used to setup forward proxy and I don't have enough knowledge on the topic to find a solution here. For me it seems like to support wildcard whole routing implementation needs to be revised from ground up... which is quite unfortunate.

**Comment by abobafett-dev** [comment]
Same for me. I have windows 11. When I want to add youtube for example I need to add 15+ IPs and for chatgpt it doesnt work and I dont know why

also when I want to add some site I need to check all IPs for example with `nslookup` command and use some methods to check which IPs site using right now (via IpvFoo for example) and its time-consuming and may not be reliable

**Comment by artemsmikh** [comment]
I agree, we really need a way to set wildcard domains and also handle possible IP changes in already added domains.

**Comment by skafiend** [comment]
Or at least, it would be great if we could black/whitelist apps

**Comment by LexBart** [comment]
The same problem. Youtube uses dozens of different subdomains.
rr2---sn-hgn7yn7l.googlevideo.com
rr9---sn-5uh5o-f5fk.googlevideo.com

It is not possible to find all ip addresses

**Comment by xcviko** [comment]
bump

**Comment by ivanglushko** [comment]
Much needed, wildcards and region based rules

**Comment by lolmaus** [comment]
PLEASE!

**Comment by momenbasel** [comment]
highly needed

**Comment by normalisht** [comment]
Still waiting

**Comment by leksand86** [comment]
> Much needed, wildcards and region based rules

region based rules that's a great idea

**Comment by lolmaus** [comment]
I've switched to `3x-ui` for the backend.

For the frontend, there's a very simple client with region-based rules `Hiddify App`, available on all platforms. In order to customize the rules, more elaborate clients are available.

Good bye, Amnezia.

**Comment by leksand86** [comment]
> I've switched to `3x-ui` for the backend.
> 
> For the frontend, there's a very simple client with region-based rules `Hiddify App`, available on all platforms. In order to customize the rules, more elaborate clients are available.
> 
> Good bye, Amnezia.

I tried to connect Hiddify to another vpn solution and it showed a lot of left connections - the application is reporting something somewhere, most likely from where and where you are connecting.

Amnesia is convenient for certain tasks, for example, for a home VPN

**Comment by FiresPOW** [comment]
very need

**Comment by Script06** [comment]
There is still no solution?

**Comment by Dimasin** [comment]
I join the request.

**Comment by rezonx3m** [comment]
Waiting this feature!

**Comment by avalanche-zxc** [comment]
agree

**Comment by MalyugaSensei** [comment]
Актуальная фича

**Comment by magnastrazh** [comment]
join

**Comment by zvnge** [comment]
bump

**Comment by Swiftkill** [comment]
> The same problem. Youtube uses dozens of different subdomains. rr2---sn-hgn7yn7l.googlevideo.com rr9---sn-5uh5o-f5fk.googlevideo.com
> 
> It is not possible to find all ip addresses

Dozens? Millions and they are dynamically changing. As well as IP for samehost can change depending on location, if that's a cloud service., for a different outgoing point it would be diffreent IPs and user cannot know that information until they connect
